File: TJCPowerPoint/PrayerRequests.vb
'This class is part of the TJC PowerPoint project.
'The TJC PowerPoint is a simple program displaying bible verses, hymns
'to a projector using Microsoft PowerPoint.
'It also has other purporses to help display usesful information to the projector
'These include:
'How to pray, service times, current time and date, holy communion slides, prayer requests
'announcements, hymnal hymns etc.
'
'Initially developed by Joshi Chan (TJC Lecieseter), from V2 onwards developed by Caleb Chan (TJC London)
'© Copyright 2023 True Jesus Church London 

Imports System.Runtime.InteropServices

Public Class PrayerRequests
    Private directory As String
    Private slideNumber As Integer

    'Method handling when prayer request form is initially loaded
    Private Sub onCreate(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.aeroEnabled = False
        Me.FormBorderStyle = FormBorderStyle.None
        PrayerRequestTxt.Focus()
        directory = MainProgram.CurrentDirectory + "\Files\PrayerRequests.txt"
        If My.Computer.FileSystem.FileExists(directory) Then
            PrayerRequestTxt.Text = IO.File.ReadAllText(directory)
        End If

        'slide number according to order of powerpoint slide
        slideNumber = MainProgram.slideDictionary.Item("prayerRequests").SlideNumber
    End Sub

    'Method handling when form is closed
    Private Sub ClosePrayerRequests_Click(sender As Object, e As EventArgs) Handles closeForm.Click
        Me.Hide()
    End Sub
    Private Sub minForm_Click(sender As Object, e As EventArgs) Handles minForm.Click
        Me.WindowState = FormWindowState.Minimized
    End Sub


    Private Sub UpdatePrayerRequests_Click(sender As Object, e As EventArgs) Handles UpdatePrayerRequests.Click
        MainProgram.ppPres.Slides(slideNumber).Shapes(1).TextFrame.TextRange.Text = PrayerRequestTxt.Text
        Try
            My.Computer.FileSystem.WriteAllText(MainProgram.CurrentDirectory + "\Files\PrayerRequests.txt", PrayerRequestTxt.Text, False)
            MessageBox.Show("Save Successful", "Save Successful")
        Catch ex As Exception
            MessageBox.Show("Save Unsuccessful", "Save Unsuccessful")
        End Try

    End Sub
    Private Sub PRFontBtn_Click(sender As Object, e As EventArgs) Handles PRFontBtn.Click
        MainProgram.ChangeFont(MainProgram.textBoxDictionary.Item("prayerRequestsTxt"))
    End Sub
    Private Sub PRColorBtn_Click(sender As Object, e As EventArgs) Handles PRColorBtn.Click
        MainProgram.ChangeColor(MainProgram.textBoxDictionary.Item("prayerRequestsTxt"))
    End Sub
    Private Sub PRTitleFontBtn_Click(sender As Object, e As EventArgs) Handles PRTitleFontBtn.Click
        MainProgram.ChangeFont(MainProgram.textBoxDictionary.Item("prayerRequestsTitle"))
    End Sub
    Private Sub PRTitleColorBtn_Click(sender As Object, e As EventArgs) Handles PRTitleColorBtn.Click
        MainProgram.ChangeColor(MainProgram.textBoxDictionary.Item("prayerRequestsTitle"))
    End Sub

    Private Sub editPrayerImage_Click(sender As Object, e As EventArgs) Handles editPrayerImage.Click
        Call MainProgram.insertPrayerImage()
    End Sub

    Private Sub deletePrayerImage_Click(sender As Object, e As EventArgs) Handles deletePrayerImage.Click
        Call MainProgram.deletePrayerImage()
    End Sub

    Private Sub goToPRSLide_Click(sender As Object, e As EventArgs) Handles goToPRSlide.Click
        MainProgram.goToPRBtn.Checked = True
    End Sub




    'FOLLOWING METHODS DEAL WITH CREATING WINDOW AS A BORDERLESS DROP SHADOW WINDOWS FORM
    'https://stackoverflow.com/questions/16493698/drop-shadow-on-a-borderless-winform#:~:text=1)%20Create%20an%20image%20having,4)%20You%20are%20done!
    Private aeroEnabled As Boolean
    Protected Overrides ReadOnly Property CreateParams() As CreateParams
        Get
            CheckAeroEnabled()
            Dim cp As CreateParams = MyBase.CreateParams
            cp.ExStyle = NativeConstants.WS_EX_COMPOSITED
            If Not aeroEnabled Then
                cp.ClassStyle = NativeConstants.CS_DROPSHADOW
                Return cp
            Else
                Return cp
            End If
        End Get
    End Property
    Protected Overrides Sub WndProc(ByRef m As Message)
        Select Case m.Msg
            Case NativeConstants.WM_NCPAINT
                Dim val = 2
                If aeroEnabled Then
                    NativeMethods.DwmSetWindowAttribute(Handle, 2, val, 4)
                    Dim bla As New NativeStructs.MARGINS()
                    With bla
                        .bottomHeight = 1
                        .leftWidth = 0
                        .rightWidth = 0
                        .topHeight = 0
                    End With
                    NativeMethods.DwmExtendFrameIntoClientArea(Handle, bla)
                End If
                Exit Select
        End Select
        MyBase.WndProc(m)
    End Sub
    Private Sub CheckAeroEnabled()
        If Environment.OSVersion.Version.Major >= 6 Then
            Dim enabled As Integer = 0
            Dim response As Integer = NativeMethods.DwmIsCompositionEnabled(enabled)
            aeroEnabled = (enabled = 1)
        Else
            aeroEnabled = False
        End If
    End Sub


    'FOLLOWING METHODS DEAL WITH FORM MOVEMENT WHEN MOUSE DRAGGED ON TOP NAV BAR
    'https://stackoverflow.com/questions/17392088/allow-a-user-to-move-a-borderless-window
    Private IsFormBeingDragged As Boolean = False
    Private MouseDownX As Integer
    Private MouseDownY As Integer
    Private Sub Form1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs) Handles navBar.MouseDown, header.MouseDown

        If e.Button = MouseButtons.Left Then
            IsFormBeingDragged = True
            MouseDownX = e.X
            MouseDownY = e.Y
        End If
    End Sub
    Private Sub Form1_MouseUp(ByVal sender As Object, ByVal e As MouseEventArgs) Handles navBar.MouseUp, header.MouseUp

        If e.Button = MouseButtons.Left Then
            IsFormBeingDragged = False
        End If
    End Sub
    Private Sub Form1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs) Handles navBar.MouseMove, header.MouseMove

        If IsFormBeingDragged Then
            Dim temp As Point = New Point()

            temp.X = Me.Location.X + (e.X - MouseDownX)
            temp.Y = Me.Location.Y + (e.Y - MouseDownY)
            Me.Location = temp
            temp = Nothing
        End If
    End Sub
End Class

File: src/TJCPowerPoint/PrayerRequests.cs
public partial class PrayerRequests
{
    private string directory;

    public PrayerRequests()
    {
        InitializeComponent();
        PrayerRequestTxt = _PrayerRequestTxt;
        _PrayerRequestTxt.Name = "PrayerRequestTxt";
    }

    private void ClosePrayerRequests_Click(object sender, EventArgs e)
    {
        this.Hide();
    }

    private void onCreate(object sender, EventArgs e)
    {
        directory = TJCPowerPoint.My.MyProject.Forms.MainProgram.Current + @"\Files\PrayerRequests.txt";
        if (System.IO.File.Exists(directory))
        {
            this.PrayerRequestTxt.Text = System.IO.File.ReadAllText(directory);
        }
    }
    private void UpdatePrayerRequests_Click(object sender, EventArgs e)
    {
        TJCPowerPoint.My.MyProject.MyForms.MainProgram.ppPres.Slides[(object)3].Shapes[(object)1].TextFrame.TextRange = this.PrayerRequestTxt.Text;
        try
        {
            TJCPowerPoint.My.MyProject.Computer.FileSystem.WriteAllText(TJCPowerPoint.My.MyProject.Forms.MainProgram.Current + @"\Files\PrayerRequests.txt", this.PrayerRequestTxt.Text, false);
            MessageBox.Show("Save Successful", "Save Successful");
        }
        catch (Exception ex)
        {
            MessageBox.Show("Save Unsuccessful", "Save Unsuccessful");
        }

    }
    private void PRFontBtn_Click(object sender, EventArgs e)
    {
        TJCPowerPoint.My.MyProject.Forms.MainProgram.ChangeFont(3, 1);
    }
    private void PRColorBtn_Click(object sender, EventArgs e)
    {
        TJCPowerPoint.My.MyProject.Forms.MainProgram.ChangeColor(3, 1);
    }
    private void PRTitleFontBtn_Click(object sender, EventArgs e)
    {
        TJCPowerPoint.My.MyProject.Forms.MainProgram.ChangeFont(3, 2);
    }
    private void PRTitleColorBtn_Click(object sender, EventArgs e)
    {
        TJCPowerPoint.My.MyProject.Forms.MainProgram.ChangeColor(3, 2);
    }

    private void editPrayerImage_Click(object sender, EventArgs e)
    {
        try
        {
            var ofd = new OpenFileDialog();
            ofd.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) + @"\Downloads";
            ofd.Filter = "Image Files(*.BMP;*.JPG;*.GIF;*.PNG)|*.BMP;*.JPG;*.GIF;*.PNG|All files (*.*)|*.*";
            if (ofd.ShowDialog() == DialogResult.OK)
            {
                TJCPowerPoint.My.MyProject.MyForms.MainProgram.ppPres.Slides[(object)2].Shapes.AddPicture(ofd.FileName, (Microsoft.Office.Core.MsoTriState)Conversions.ToInteger(false), (Microsoft.Office.Core.MsoTriState)Conversions.ToInteger(true), 0f, 0f, TJCPowerPoint.My.MyProject.MyForms.MainProgram.ppPres.PageSetup.SlideWidth, TJCPowerPoint.My.MyProject.MyForms.MainProgram.ppPres.PageSetup.SlideHeight);
                System.IO.File.WriteAllText(TJCPowerPoint.My.MyProject.Forms.MainProgram.Current + @"\Files\prayerImgDir.txt", ofd.FileName);
                MessageBox.Show("Prayer Image Was Successfully Updated", "Success");
            }
            else
            {
                MessageBox.Show("Prayer Image Was Not Successfully Updated. Please Try Again", "Error");
            }
        }


        catch (Exception ex)
        {
            MessageBox.Show("Prayer Image Was Not Successfully Updated. Please Try Again", "Error");
        }

    }

    private void Button1_Click(object sender, EventArgs e)
    {
        TJCPowerPoint.My.MyProject.MyForms.MainProgram.ppPres.SlideShowWindow.View.GotoSlide(3);
    }


}
